### PR TITLE
Windows: Resolve path to lib.exe (for static libs) at reggae-time

### PR DESCRIPTION
**Pull request by kinke**
Otherwise, a set-up MSVC environment is required for the build too, not just the reggae invocation. cl.exe is already resolved to an absolute path.

### CODE REVIEW

**Comment by kinke** [comment]
Oh god, this is what's currently happening:

* `build.ninja`:
  ```
  build mylib.lib: Program .reggae\objs\mylib.lib.objs\csrc\foo.obj
    before = Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.16.27023\bin\Hostx64\x64\lib.exe" /OUT:
  ```
* `rules.ninja`:
  ```
  rule Program
    command = "C:\Program $before$out $in
  ```

**Comment by kinke** [comment]
Ready now.

File: payload/reggae/rules/common.d [hunk 1]
```diff
@@ -487,29 +487,38 @@ Target staticLibrary(
     );
 }
 
-Target staticLibraryTarget(in string name, Target[] objects) @safe pure {
+Target staticLibraryTarget(in string name, Target[] objects) @safe {
     import std.path: defaultExtension;
     return Target(
         [buildPath("$builddir", defaultExtension(name, libExt))],
-        staticLibraryShellCommand,
+        staticLibraryShellCommand(),
```
Comment (atilaneves) [inline payload/reggae/rules/common.d:494]: ?

File: payload/reggae/rules/common.d [hunk 2]
```diff
@@ -487,29 +487,38 @@ Target staticLibrary(
...
     );
 }
 
-Target staticLibraryTarget(in string name, Target[] objects, Target[] implicits) @safe pure {
+Target staticLibraryTarget(in string name, Target[] objects, Target[] implicits) @safe {
     import std.path: defaultExtension;
     return Target(
         [buildPath("$builddir", defaultExtension(name, libExt))],
-        staticLibraryShellCommand,
+        staticLibraryShellCommand(),
```
Comment (atilaneves) [inline payload/reggae/rules/common.d:503]: ?

File: payload/reggae/rules/common.d (hunk 1 above)
Comment (kinke) [reply payload/reggae/rules/common.d:494]: Heh, personal preference (remember DConf last year?) - to make clear this needs to *run*, isn't CTFE-able (on Windows).

**Review by atilaneves — APPROVED** [review]

**Pull request merged**